### PR TITLE
[cpuinfo,fbgemm,xnnpack] update cpuinfo to latest

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?

With https://github.com/pytorch/cpuinfo/pull/69, `cpuinfo` now supports CMake `find_package`.
Remove the `unofficial::` prefix from consumer ports.

#### Affected Ports

* nnpack #16344
* fbgemm #17063
* xnnpack #17064
* qnnpack #17065

### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?

The current version didn't support the following triplets.

#### Not supported
* `arm64-windows`

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

**Comment by FrankXie05** [comment]
cc @JackBoosY Could you help review it? Thanks in advance.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
I'm still worried about the new support expressions, is there any evidence that they didn't support them before?

**Comment by luncliff** [comment]
> I'm still worried about the new support expressions, is there any evidence that they didn't support them before?

Looks like this is the source of the problem. Only x86 implementation for Windows.

https://github.com/pytorch/cpuinfo/blob/b40bae27785787b6dd70788986fd96434cf90ae2/CMakeLists.txt#L144-L146

**Comment by JackBoosY** [comment]
> Looks like this is the source of the problem. Only x86 implementation for Windows.

Does this prove that x86 source code can be used to build under x64?

> The ARM folder doesn't contain a Windows platform folder. :(

However, there is a arm api header: https://github.com/pytorch/cpuinfo/blob/5916273f79a21551890fd3d56fc5375a78d1598d/src/arm/api.h

**Comment by luncliff** [comment]
Maybe. I think the x86 source is compatible with x64.
I'm not sure the linux init.c source can be used for Windows.

**Comment by JackBoosY** [comment]
@luncliff You can remove the supports field and see what happened in pipeline tests.

**Comment by luncliff** [comment]
It always select x86 implementation in Windows. Current `master` branch's `cpuinfo` port will fail will identifier errors.

File: ports/cpuinfo/vcpkg.json [hunk 1]
```diff
@@ -1,8 +1,20 @@
 {
   "name": "cpuinfo",
-  "version-date": "2021-04-04",
+  "version-date": "2022-04-02",
   "description": "CPU INFOrmation library (x86/x86-64/ARM/ARM64, Linux/Windows/Android/macOS/iOS)",
   "homepage": "https://github.com/pytorch/cpuinfo",
+  "license": "BSD-2-Clause",
+  "supports": "!(arm & windows) | (arm & uwp)",
```
Comment (JackBoosY) [inline ports/cpuinfo/vcpkg.json:7]: cpuinfo:arm64-windows also fails to build in master branch.

**Review by JackBoosY — APPROVED** [review]

File: ports/cpuinfo/vcpkg.json (hunk 1 above)
Comment (strega-nil-ms) [reply ports/cpuinfo/vcpkg.json:7]: fwiw, I can build it on my machine as `arm64-windows`, so I think this supports is wrong.

**Comment by strega-nil-ms** [comment]
Since I can build it on my machine as both `arm64-windows` and `arm-windows`, I believe the supports is wrong; let's see what CI says.

**Comment by luncliff** [comment]
> It looks like the issue is that cpuinfo is broken on x64 targeting arm64; given that it can still build on arm64 targeting arm64, I do not want to put that in supports; it'd be better to fix it for real.
> 

I will wait for this. Removing the part can be done easily.

**Comment by strega-nil-ms** [comment]
WFH again today, so I'll take a look tomorrow (sorry!)

**Pull request merged**